core/tests/ts-integration/tests/custom-account.test.ts

```diff
@@ -95,6 +113,94 @@ describe('Tests for the custom account behavior', () => {
         ).toBeAccepted([erc20BalanceChange, feeCheck]);
     });
 
+    test('Should fail transaction validation due to timestamp assertion in the validation tracer - short time window', async () => {
+        const now = Math.floor(Date.now() / 1000);
+        const rangeStart = now - 60;
+        const rangeEnd = now + 60;
+
+        const customAccount = await deployAndFundCustomAccount(
+            alice,
+            erc20Address,
+            timestampAsserterAddress,
+            rangeStart,
+            rangeEnd
+        );
+
+        const tx = await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT);
+        await expect(
+            sendCustomAccountTransaction(
+                tx as zksync.types.Transaction,
+                alice.provider,
+                await customAccount.getAddress(),
+                testMaster.environment().l2ChainId
+            )
+        ).toBeRejected(
+            'failed to validate the transaction. reason: Violated validation rules: block.timestamp range is too short'
+        );
+    });
+
+    test('Should fail transaction validation due to timestamp assertion in the validation tracer - close to the range end', async () => {
+        const now = Math.floor(Date.now() / 1000);
+        const rangeStart = now - 600;
+        const rangeEnd = now + 30;
+
+        const customAccount = await deployAndFundCustomAccount(
+            alice,
+            erc20Address,
+            timestampAsserterAddress,
+            rangeStart,
+            rangeEnd
+        );
+
+        const tx = await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT);
+
+        await expect(
+            sendCustomAccountTransaction(
+                tx as zksync.types.Transaction,
+                alice.provider,
+                await customAccount.getAddress(),
+                testMaster.environment().l2ChainId
+            )
+        ).toBeRejected(
+            'failed to validate the transaction. reason: Violated validation rules: block.timestamp is too close to the range end'
+        );
+    });
+
+    test('Should fail to estimate fee due to block.timestamp assertion in the smart contract', async () => {
+        const now = Math.floor(Date.now() / 1000);
+        const rangeStart = now + 300;
+        const rangeEnd = now + 1000;
+
+        const customAccount = await deployAndFundCustomAccount(
+            alice,
+            erc20Address,
+            timestampAsserterAddress,
+            rangeStart,
+            rangeEnd
+        );
+
+        const promise = alice.provider.estimateGas({
+            ...(await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT)),
+            from: await customAccount.getAddress()
+        });
+
+        // requires custom assertion because the transaction would fail in "estimateFee" that is not handled correctly by zksync ethers
```

Comment:
Dumb question: Would it be possible to use another RPC method then? (E.g., submit a transaction w/o estimating gas for it.)

Reply:
You'd still have to fill all required fields though no? Even if you call 'sendRawTransaction` without estimating fee first.

Reply:
Yes, but I thought that in this case it would be possible to simplify error assertions below similarly to other tests.

Reply:
Very good point - changed the test